database/upload.go

```diff
@@ -80,10 +82,64 @@ func (db *DB) UploadsBySkylink(ctx context.Context, skylink Skylink, offset, pag
 	if err := validateOffsetPageSize(offset, pageSize); err != nil {
 		return nil, 0, err
 	}
-	matchStage := bson.D{{"$match", bson.D{{"skylink_id", skylink.ID}}}}
+	matchStage := bson.D{{"$match", bson.D{
+		{"skylink_id", skylink.ID},
+		{"unpinned", false},
+	}}}
 	return db.uploadsBy(ctx, matchStage, offset, pageSize)
 }
 
+// UploadsBySkylinkAndUser fetches all uploads of this skylink by this user.
+func (db *DB) UploadsBySkylinkAndUser(ctx context.Context, skylink Skylink, user User) ([]Upload, error) {
```

Comment:
Is this used anywhere?

Reply:
Hm, it doesn't seem so. I guess it was and then I forgot to remove it. I'll clean it up now.